torchaudio_contrib/layers.py

```diff
@@ -1,142 +1,241 @@
+
+    Args:
+
+        rate (float): rate to speed up or slow down by.
+        hop_length (int): Number audio of frames between STFT columns.
+            Defaults to 512.
+        n_stft (int, optional): number of filter banks from stft.
+            Defaults to 1025.
+    """
+
+    def __init__(self, rate, hop_length=512, n_stft=1025):
+
+        self.rate = rate
+
+        phi_advance = torch.linspace(
+            0, math.pi * hop_length, n_stft)[..., None]
+
+        super(StretchSpecTime, self).__init__({'phi_advance': phi_advance})
 
     def forward(self, x):
-        """
-        Input Tensor shape -> (batch, channel, signal)
-        Output Tensor shape -> (batch, channel, mel_freq, time)
-        """
+        return phase_vocoder(x, self.rate, self.phi_advance)
+
+    def __repr__(self):
+        param_str = '(rate={})'.format(self.rate)
+        return self.__class__.__name__ + param_str
+
+
+def Spectrogram(n_fft=2048, hop_length=None, len_win=None,
+                window=None, pad=0, pad_mode="reflect", power=1., **kwargs):
+    """
+    Get spectrogram module.
 
-        spec = super(Melspectrogram, self).forward(x)
-        spec = torch.matmul(spec.transpose(2, 3), self.mel_fb).transpose(2, 3)
-        return spec
+    Args:
+
+        n_fft (int): FFT window size. Defaults to 2048.
+        hop_length (int): Number audio of frames between STFT columns.
+            Defaults to n_fft // 4.
+        len_win (int): Size of stft window. Defaults to n_fft.
+        window (Tensor): 1-D tensor.
+            Defaults to Hanning Window of size len_win.
+        pad (int): Amount of padding to apply to signal. Defaults to 0.
+        pad_mode: padding method (see torch.nn.functional.pad).
+            Defaults to "reflect".
+        power (float): What power to normalize to. Defaults to 1.
+        **kwargs: Other torch.stft parameters, see torch.stft for more details.
+    """
+    return nn.Sequential(
+        STFT(
+            n_fft,
+            hop_length,
+            len_win,
+            window,
+            pad,
+            pad_mode,
+            **kwargs),
+        ComplexNorm(power))
+
+
+def Melspectrogram(n_mels=128, sr=44100, f_min=0.0,
+                   f_max=None, n_stft=None, **kwargs):
+    """
+    Get melspectrogram module.
+
+    Args:
+        n_mels (int): number of mel bins. Defaults to 128.
+        sr (int): sample rate of audio signal. Defaults to 44100.
+        f_max (float, optional): maximum frequency. Defaults to sr // 2.
+        f_min (float): minimum frequency. Defaults to 0.
+        n_stft (int, optional): number of filter banks from stft.
+            Defaults to n_fft//2 + 1 if 'n_fft' in kwargs else 1025.
+        **kwargs: torchaudio_contrib.Spectrogram parameters.
+    """
+    n_fft = kwargs.get('n_fft', None)
+    n_stft = n_fft // 2 + 1 if n_fft else n_stft
+    return nn.Sequential(*Spectrogram(**kwargs),
+                         MelFilterbank(n_mels, sr, f_min, f_max, n_stft))
```

Comment:
Since I'm unpacking `nn.Sequential` this won't happen, see:
```python
In [2]: nn.Sequential(*nn.Sequential(nn.Module(), nn.Module()), nn.Module())
Out[2]: 
Sequential(
  (0): Module()
  (1): Module()
  (2): Module()
)
```

Reply:
Oh, neat, I didn't know `Sequential` was iterable (and I overlooked the `*`)!